### PR TITLE
Added Commit Reward tests

### PR DESCRIPTION
**Pull request by tanya-saroha**
Added commit reward test cases to see if transactions are being created.
Commit reward file for now has random score and reward generation.

### CODE REVIEW

File: test/models/commit_reward_test.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    assert_not_equal @user1.points, nil
+    assert_not_equal @user2.points, nil
+    
+    assert_not_equal @user1.transactions.first.type, nil
+
+    assert_not_equal @user2.transactions.last.transaction_type, nil
+
+    assert_not_equal @user1.commits.sum{|commit| commit.score}, nil
+
+    assert_not_equal @user1.badges, {}
```
Comment (pratik14) [inline test/models/commit_reward_test.rb]: hey what are to testing here, if everything is nil

File: test/models/commit_reward_test.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+require 'test_helper'
+
+class CommitRewardTest < ActiveSupport::TestCase
+  def setup
+    @user1  = create :user, :auth_token => 'abcd1234'
```
Comment (pratik14) [inline test/models/commit_reward_test.rb]: ```create :user, auth_token: 'abcd1234', name: 'user1'```

File: test/models/commit_reward_test.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    @user1.name = 'user1'
+
+    @user2 = create :user, :auth_token => 'abc1234'
+    @user2.name = 'user2'
+
+    @repository1 = create :repository, :language => 'Ruby'
+    @repository2 = create :repository, :language => 'Java'
+
+    3.times.each do |i|
```
Comment (pratik14) [inline test/models/commit_reward_test.rb]: ``` create_list create :commit, 3 user: @user1, commit_date: Date.yesterday, repository: @repository1, score: 2, reward: 1 ```

File: test/models/commit_reward_test.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    end
+
+    2.times.each do |i|
+      commit = create :commit, user: @user2, commit_date: Date.yesterday, repository: @repository2, score: 2.4, reward: 1.6
+      @user2.commits << commit
+    end
+  end
+
+  test 'should check if transactions are created properly' do
+    CommitReward.new.send(:create_transaction)
```
Comment (pratik14) [inline test/models/commit_reward_test.rb]: call this ``` CommitReward.new.calculate```

Comment (tanya-saroha) [reply test/models/commit_reward_test.rb]: If I do CommitReward.new.calculate, random values will get assigned to score and reward and I wont be able to check them the way I have checked now.
 Here we just want to check if transaction is being created porperly and badges are updated or not, hence I only called that method.

File: test/models/commit_reward_test.rb (hunk 2 above)
Comment (tanya-saroha) [reply test/models/commit_reward_test.rb]: Done

File: test/models/commit_reward_test.rb (hunk 3 above)
Comment (tanya-saroha) [reply test/models/commit_reward_test.rb]: Done

**Pull request merged**